docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -0,0 +1,469 @@
+&#xfeff;                    | activation         | (categorical) | An identifier for each layer
+&#xfeff;                    |                    | tanh          | hyperbolic tangent
+&#xfeff;                    |                    | sigmoid       | sigmoid function
+&#xfeff;                    |                    | relu          | relu function
+&#xfeff;                    |                    | softmax       | softmax function
+&#xfeff;                    | weight_initializer | (categorical) | Weight initializer
+&#xfeff;                    |                    | zeros         | Zero initialization
+&#xfeff;                    |                    | lecun_normal  | LeCun Normal Initialization
+&#xfeff;                    |                    | lecun_uniform | LeCun Uniform Initialization
+&#xfeff;                    |                    | xavier_normal | Xavier Normal Initialization
+&#xfeff;                    |                    | xavier_uniform| Xavier Uniform Initialization
+&#xfeff;                    |                    | he_normal     | He Normal Initialization
+&#xfeff;                    |                    | he_uniform    | He Uniform Initialization
+&#xfeff;                    | bias_initializer   | (categorical) | Bias initializer, same category as `weight_initializer`
+&#xfeff;                    | weight_regularizer | (categorical) | Weight regularizer. Currently, only l2norm is supported
+&#xfeff;                    |                    | l2norm        | l2norm weight regularizer
+&#xfeff;                    | weight_regularizer_constant | (float)| weight regularizer constant
+&#xfeff;                    | flatten            | (boolean)     | flatten shape to c:h:w -> 1:1:c*h*w
+`ML_TRAIN_LAYER_TYPE_INPUT` |                    |               | Input layer
+&#xfeff;                    | normalization      | (boolean)     | normalize input if true
+&#xfeff;                    | standardization    | (boolean)     | standardize input if true
+`ML_TRAIN_LAYER_TYPE_FC`    |                    |               | Fully connected layer
+&#xfeff;                    | unit               | (integer)     | number of outputs
+
+### Optimizer
+
+Optimizer determines how to update model parameters according to loss from prediction.
+Currently, Stochastic Gradient Descent optimizer and Adam optimizer are supported:
+
+```c
+// Create optimizer
+ml_train_optimizer_h optimizer;
+ml_train_optimizer_create(&optimizer, ML_TRAIN_OPTIMIZER_TYPE_SGD);
+
+// configure optimizer
+ml_train_optimizer_set_property(optimizer, "learning_rate=0.001", NULL);
+
+// after configuring optimizer to model
+// you do not need to destroy optimizer since ownership is transferred to the model.
+ml_train_model_set_optimizer(model, optimizer);
```

Comment:
Changed for clarity. Please review this version if possible :)

```suggestion
// Set optimizer to the model
// don't need to destroy optimizer after setting optimizer
// since ownership is transferred to the model.
ml_train_model_set_optimizer(model, optimizer);
```